pandas/tests/test_indexing.py

```diff
@@ -5619,19 +5650,12 @@ def f():
                           slice(3, 4.0),
                           slice(3.0, 4.0)]:
 
-                    def f():
-                        s.iloc[l]
-                    self.assertRaises(TypeError, f)
-
-                    def f():
-                        s.loc[l]
-                    self.assertRaises(TypeError, f)
+                    for idxr in [lambda x: x.loc,
+                                 lambda x: x.iloc,
+                                 lambda x: x]:
+                        self.assertRaises(TypeError, lambda: idxr(s)[l])
```

Comment:
I am probably missing something (the diff in the tests is a bit hard to follow), but shouldn't there be a difference between loc and iloc here? As loc is doing label-based slicing, so float is allowed?